epcis_event_hash_generator/xml_to_py.py

```diff
@@ -124,3 +128,28 @@ def event_list_from_epcis_document_xml(path):
     logging.debug("Simple python object:\n%s", obj)
 
     return obj
+
+
+def event_list_from_epcis_document_xml_str(xmlStr):
+    """
+    Read EPCIS XML document and generate the event List in the form of a simple python object
+    """
+    try:
+        data = removeExtensionTags(xmlStr.decode("utf-8"))
+
+        root = ElementTree.fromstring(data)
+
+        eventList = root.find("*EventList")
+        if not eventList:
+            raise ValueError("No EventList found")
+    except (ValueError, OSError) as ex:
+        logging.debug(ex)
+        logging.error("Input string does not contain a valid EPCIS XML document with EventList.")
+        return ("", "", [])
+
+    # sort=False => preserve document order of events
+    obj = _xml_to_py(eventList, False)
+
+    logging.debug("Simple python object:\n%s", obj)
+
+    return obj
```

Comment:
this is a combination of code in `event_list_from_epcis_document_xml` and `_read_epcis_document_xml`. Could you make sure that this code is not duplicated? E.g. replace the upper to functions by
 
```
def event_list_from_epcis_document_xml(path):
    with open(path, 'r') as file:
        data = file.read()

    return event_list_from_epcis_document_xml_str(data)
```

Reply:
absolutely! have removed the duplication as suggested.